cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -261,8 +262,8 @@ public ServiceTicket grantServiceTicket(
             }
             ticketGrantingTicket.getSupplementalAuthentications().add(currentAuthentication);
         }
-        
-        if (!registeredService.isSsoEnabled() && currentAuthentication == null) {
+
+        if (currentAuthentication == null && !registeredService.getAuthorizationStrategy().isServiceAuthorizedForSso()) {

```

Comment:
I still think its weird to mix service options with an authorization strategy for people to use it.  Curious as to what others think.